test/routeBuilderSpec.js

```diff
@@ -0,0 +1,286 @@
+'use strict'
+
+const assert = require('assert')
+const RoutingTables = require('five-bells-connector')._test.RoutingTables
+const RouteBuilder = require('five-bells-connector')._test.RouteBuilder
+
+const baseURI = 'http://mark.example'
+const ledgerA = 'http://ledgerA.example'
+const ledgerB = 'http://ledgerB.example'
+const ledgerC = 'http://ledgerC.example'
+
+// sending/receiving users
+const aliceA = 'http://ledgerA.example/accounts/alice'
+const bobB = 'http://ledgerB.example/accounts/bob'
+const carlC = 'http://ledgerC.example/accounts/carl'
+
+// connector users
+const markA = 'http://ledgerA.example/accounts/mark'
+const markB = 'http://ledgerB.example/accounts/mark'
+const maryB = 'http://ledgerB.example/accounts/mary'
+
+describe('RouteBuilder', function () {
+  beforeEach(function * () {
+    this.precisionCache = {
+      get: function * (ledger) {
+        return {precision: 10, scale: 2}
+      }
+    }
+
+    this.tables = new RoutingTables(baseURI, [{
+      source_ledger: ledgerA,
+      destination_ledger: ledgerB,
+      connector: 'http://mark.example',
+      source_account: markA,
+      destination_account: markB,
+      min_message_window: 1,
+      points: [ [0, 0], [200, 100] ]
+    }])
+    this.builder = new RouteBuilder(this.tables, this.precisionCache, {
+      minMessageWindow: 1,
+      slippage: 0.01,
+      ledgerCredentials: {
+        'http://ledgerA.example': {account_uri: markA},
+        'http://ledgerB.example': {account_uri: markB}
+      }
+    })
+  })
+
+  describe('getQuote', function () {
+    beforeEach(function * () {
+      this.tables.addRoute({
+        source_ledger: ledgerB,
+        destination_ledger: ledgerC,
+        connector: 'http://mary.example',
+        source_account: maryB,
+        min_message_window: 1,
+        points: [ [0, 0], [200, 100] ]
+      })
+    })
+
+    describe('fixed sourceAmount', function () {
+      it('returns a quote with slippage in the final amount', function * () {
+        const quoteTransfer = yield this.builder.getQuote({
+          sourceLedger: ledgerA,
+          sourceAccount: aliceA,
+          destinationLedger: ledgerC,
+          destinationAccount: carlC,
+          sourceAmount: '100'
+        })
+        assert.deepStrictEqual(quoteTransfer, {
+          ledger: ledgerA,
+          debits: [{ account: aliceA, amount: '100.00' }],
+          credits: [{
+            account: markA,
+            amount: '100.00',
+            memo: {
+              destination_transfer: {
+                ledger: ledgerC,
+                debits: [{ account: null, amount: '24.75' }],
+                credits: [{ account: carlC, amount: '24.75' }]
+              }
+            }
+          }],
+          _hop: quoteTransfer._hop
+        })
+      })
+    })
```

Comment:
Yes, see [`getCheaperQuote`](https://github.com/interledger/five-bells-sender/pull/65/files#diff-795d9d410a3d31edd97b5216a81441baR41).

Regardless of what non-5B transfer object the connector returns, the sender needs to understand how to compare them.

Maybe it could end up as part of the multiledger abstraction.